addons/settings/fnc_parse.sqf

```diff
@@ -13,65 +13,58 @@ Returns:
-    // Means that strings inside arrays don't support white space chars, but w/e.
-    // No such setting exists atm anyway.
-    if (_string find """" != 0) then {
-        _string = _string splitString WHITESPACE joinString "";
-    };
-
-    parseSimpleArray (["[", _string, "]"] joinString "") select 0
-};
-
 params [["_info", "", [""]], ["_validate", false, [false]], ["_source", "", [""]]];
 
-// remove whitespace at start and end of each line
-private _result = [];
+// Remove whitespace at start and end of each line, a line being define by the ";" at its end
+private _parsed = [];
 
 {
-    _result pushBack (_x call CBA_fnc_trim);
-} forEach (_info splitString NEWLINE);
+    _parsed pushBack (trim _x);
+} forEach (_info splitString ";");
 
-{
-    if (_x select [count _x - 1] != ";") then {
-        _result set [_forEachIndex, _x + ";"];
-    };
-} forEach _result;
+// Remove empty strings
+_parsed = _parsed - [""];
 
-_info = (_result joinString NEWLINE) + NEWLINE;
-
-// separate statements (setting = value)
-_result = [];
+// Separate statements (setting = value)
+private _result = [];
+private _indexEqualSign = -1;
+private _setting = "";
+private _priority = 0;
+private _value = "";
+private _countForce = count "force";
+private _whitespaces = [toString [ASCII_NEWLINE], toString [ASCII_CARRIAGE_RETURN], toString [ASCII_TAB], toString [ASCII_SPACE]];
 
 {
-    private _indexEqualSign = _x find "=";
+    _indexEqualSign = _x find "=";
 
-    private _setting = (_x select [0, _indexEqualSign]) call CBA_fnc_rightTrim;
-    private _priority = 0;
+    // Setting name is front of "="
+    _setting = (_x select [0, _indexEqualSign]) trim [WHITESPACE, 2];
+    _priority = 0;
 
-    if (_setting select [0, count "force"] == "force") then {
-        _setting = _setting select [count "force"] call CBA_fnc_leftTrim;
+    // Check if the first entry is force and not followed by an empty space
+    if (_setting select [0, _countForce] == "force" && {(_setting select [_countForce, 1]) in _whitespaces}) then {
+        _setting = (_setting select [_countForce]) trim [WHITESPACE, 1];
         _priority = _priority + 1;
     };
 
-    if (_setting select [0, count "force"] == "force") then {
-        _setting = _setting select [count "force"] call CBA_fnc_leftTrim;
+    // Check if the second entry is force and not followed by an empty space
+    if (_setting select [0, _countForce] == "force" && {(_setting select [_countForce, 1]) in _whitespaces}) then {
+        _setting = (_setting select [_countForce]) trim [WHITESPACE, 1];
         _priority = _priority + 1;
     };
 
+    // If setting is valid, get it's value
     if (_setting != "") then {
-        private _value = ((_x select [_indexEqualSign + 1]) call CBA_fnc_trim) call _fnc_parseAny;
+        // Remove whitespaces; Parse bool, number, string
+        _value = parseSimpleArray (["[", (_x select [_indexEqualSign + 1]) splitString WHITESPACE joinString "", "]"] joinString "") select 0;
```

Comment:
idk, feels to me like this should be broken up.

Reply:
Reverted back to original formatting.